Refit/RequestBuilderImplementation.cs

```diff
@@ -244,13 +244,13 @@ async Task AddMultipartItemAsync(MultipartFormDataContent multiPartContent, stri
                 {
                     resp = await client.SendAsync(rq, HttpCompletionOption.ResponseHeadersRead, ct).ConfigureAwait(false);
                     content = resp.Content ?? new StringContent(string.Empty);
-                    ApiException e = null;
+                    Exception e = null;
                     disposeResponse = restMethod.ShouldDisposeResponse;
 
                     if (!resp.IsSuccessStatusCode && typeof(T) != typeof(HttpResponseMessage))
                     {
                         disposeResponse = false; // caller has to dispose
-                        e = await ApiException.Create(rq, restMethod.HttpMethod, resp, restMethod.RefitSettings).ConfigureAwait(false);
+                        e = await restMethod.ErrorHandler.HandleErrorAsync(rq, restMethod.HttpMethod, resp, restMethod.RefitSettings).ConfigureAwait(false);
```

Comment:
What happens if the error handler returns null?

Reply:
Now it will go the way of deserializing the response and the Refit code will return `default` for this type.

It is possible to alter on when the user implementation considers that this is not an error scenario, they can return null and the refit code can deserialize the response.

For example:
Normal response
```json
{
    "result": true,
    "data" : { ... }
}
```
Error response
```json
{
    "result": false,
    "error": "error description"
}
```

and model for this
```c#
public class ApiMethodResponse
{
    public bool Result { get; set; }
    public string Error { get; set; }
    public SomeDataModel Data { get; set; }
}
```